### ISSUE TITLE
navigo router: constructAbsoluteUrl() ignores "useHash" option

### ISSUE THREAD

**Issue by jpommerening** [LaxarJS/laxar#440]
Yo!

`navigo.link()` does not prepend a hash. With `useHash: true` and the default hash-symbol, the [following function](https://github.com/LaxarJS/laxar/blob/f67a03593adab18c4bf141750ca82f234bb4eace/lib/runtime/navigo_router.js#L190) returns things like `http://localhost:8500/debug.html/my-place`:

```js
function constructAbsoluteUrl( patterns, parameters ) {
   return router.link( constructPath( patterns, parameters ) );
}
```

We should probably replace it with

```js
function constructAbsoluteUrl( patterns, parameters ) {
   const hashPrefix = useHash ? hash : '';
   return router.link( hashPrefix + constructPath( patterns, parameters ) );
}
```

fyi @alex3683

**Comment by x1B** [comment]
Pretty weird that `navigo.link` does not use the hash in the first place!

**Comment by jpommerening** [comment]
Yup. Another option would be to use [`navigo.generate`](https://github.com/krasimir/navigo/blob/master/src/index.js#L287) in `constructPath` but that would have taken more than 5 minutes. 😅

**Comment by alex3683** [comment]
I used to use `navigo.generate`, but it used to include a hash, that would break when using the resulting path with navigate, since navigate added another hash. Additionally we needed our special query parameter handling, that would have needed some code anyways. Since that code was already available in the pagejs router, I simply copied it to the navigo router 😎
Another thing why ignoring `navigo.generate` was a good thing to do is this breaking change in a bugfix version introduced four days ago: https://github.com/krasimir/navigo/issues/87

Seems that while switching from `generate` to the pagejs router code I missed the missing hash. As @x1B already pointed out APIs of `link` and `navigate` are totally inconsistent regarding the handling of their respective `path` argument.

I don't like [NIH](https://en.wikipedia.org/wiki/Not_invented_here) stuff, but it seems finding a basic routing lib that gets out of your way and works is pretty hard ...

Will fix this soonish.

**Comment by alex3683** [comment]
Since they don't seem to take semver that seriously, I'll fix navigo to version `4.3.6`.

**Comment by alex3683** [comment]
Fixed on `master` (`v2.0.0`).

**Comment by jpommerening** [comment]
Thanks! 🙏
